### PR TITLE
Stop(deallocate) and Start vm actions

### PR DESCRIPTION
**Pull request by sumeetisp**
Adding missing functionality for starting and stopping vm

### What does this PR do?
This PR adds functionality for stopping(deallocate) and starting azure virtual machines.

### Previous Behavior
Remove this section if not relevant

### New Behavior
Remove this section if not relevant

### Tests written?

Yes/No

### Commits signed with GPG?

Yes/No

Please review [Salt's Contributing Guide](https://docs.saltstack.com/en/latest/topics/development/contributing.html) for best practices.

See GitHub's [page on GPG signing](https://help.github.com/articles/signing-commits-using-gpg/) for more information about signing commits with GPG.

### CODE REVIEW

**Comment by rallytime** [comment]
Hi @sumeetisp! Thank you for this addition. Can you fix the lint errors https://jenkins.saltstack.com/job/PR/job/salt-pr-lint-n/17562/violations/file/salt/cloud/clouds/azurearm.py/?

**Comment by sumeetisp** [comment]
Done

File: salt/cloud/clouds/azurearm.py [hunk 1]
```diff
@@ -1834,3 +1834,63 @@ def delete_blob(call=None, kwargs=None):  # pylint: disable=unused-argument
 
     storageservice.delete_blob(kwargs['container'], kwargs['blob'])
     return True
+
+
+def stop(name, resource_group=None, call=None):
+    '''
+    Stop a VM
+
```
Comment (rallytime) [inline salt/cloud/clouds/azurearm.py]: Can you add a `.. versionadded:: Fluorine` here?

File: salt/cloud/clouds/azurearm.py [hunk 2]
```diff
@@ -1834,3 +1834,63 @@ def delete_blob(call=None, kwargs=None):  # pylint: disable=unused-argument
...
+    else:
+        instance_stop = compconn.virtual_machines.stop(resource_group, name)
+        instance_stop.wait()
+    return instance_stop
+
+
+def start(name, resource_group=None, call=None):
+    '''
+    Start a VM
+
```
Comment (rallytime) [inline salt/cloud/clouds/azurearm.py]: This also need the same version added tag.

File: salt/cloud/clouds/azurearm.py [hunk 3]
```diff
@@ -1834,3 +1834,63 @@ def delete_blob(call=None, kwargs=None):  # pylint: disable=unused-argument
...
+            'The stop action must be called with -a or --action.'
+        )
+    if not compconn:
+        compconn = get_conn()
+    if resource_group is None:
+        for group in list_resource_groups():
+            try:
+                instance_stop = compconn.virtual_machines.deallocate(group, name)
+                instance_stop.wait()
+                resource_group = group
```
Comment (rallytime) [inline salt/cloud/clouds/azurearm.py]: What is the purpose of setting this resource_group variable? It looks like it's not being used anywhere later.

File: salt/cloud/clouds/azurearm.py [hunk 4]
```diff
@@ -1834,3 +1834,63 @@ def delete_blob(call=None, kwargs=None):  # pylint: disable=unused-argument
...
+            'The start action must be called with -a or --action.'
+        )
+    if not compconn:
+        compconn = get_conn()
+    if resource_group is None:
+        for group in list_resource_groups():
+            try:
+                instance_start = compconn.virtual_machines.start(group, name)
+                instance_start.wait()
+                resource_group = group
```
Comment (rallytime) [inline salt/cloud/clouds/azurearm.py]: Same question about setting the resource group here as above.

**Review by rallytime — CHANGES_REQUESTED** [review]
I had a couple of questions and suggestions here. Otherwise this looks good.

**Comment by sumeetisp** [comment]
Hi Nicole,

I will be reviewing the code after getting back from vacation on 1st Jan.

Thanks,
Sumeet Salvankar

On 22-Dec-2017 23:17, "Nicole Thomas" <notifications@github.com> wrote:

*@rallytime* requested changes on this pull request.

I had a couple of questions and suggestions here. Otherwise this looks good.
------------------------------

In salt/cloud/clouds/azurearm.py
<https://github.com/saltstack/salt/pull/45119#discussion_r158530961>:

> @@ -1834,3 +1834,63 @@ def delete_blob(call=None, kwargs=None):  # pylint: disable=unused-argument

     storageservice.delete_blob(kwargs['container'], kwargs['blob'])
     return True
+
+
+def stop(name, resource_group=None, call=None):
+    '''
+    Stop a VM
+

Can you add a .. versionadded:: Fluorine here?
------------------------------

In salt/cloud/clouds/azurearm.py
<https://github.com/saltstack/salt/pull/45119#discussion_r158531114>:

> +            try:
+                instance_stop =
compconn.virtual_machines.deallocate(group, name)
+                instance_stop.wait()
+                resource_group = group
+            except CloudError:
+                continue
+    else:
+        instance_stop = compconn.virtual_machines.stop(resource_group, name)
+        instance_stop.wait()
+    return instance_stop
+
+
+def start(name, resource_group=None, call=None):
+    '''
+    Start a VM
+

This also need the same version added tag.
------------------------------

In salt/cloud/clouds/azurearm.py
<https://github.com/saltstack/salt/pull/45119#discussion_r158531258>:

> +    .. code-block:: bash
+
+         salt-cloud -a stop myminion
+    '''
+    if call == 'function':
+        raise SaltCloudSystemExit(
+            'The stop action must be called with -a or --action.'
+        )
+    if not compconn:
+        compconn = get_conn()
+    if resource_group is None:
+        for group in list_resource_groups():
+            try:
+                instance_stop =
compconn.virtual_machines.deallocate(group, name)
+                instance_stop.wait()
+                resource_group = group

What is the purpose of setting this resource_group variable? It looks like
it's not being used anywhere later.
------------------------------

In salt/cloud/clouds/azurearm.py
<https://github.com/saltstack/salt/pull/45119#discussion_r158531323>:

> +    .. code-block:: bash
+
+         salt-cloud -a start myminion
+    '''
+    if call == 'function':
+        raise SaltCloudSystemExit(
+            'The start action must be called with -a or --action.'
+        )
+    if not compconn:
+        compconn = get_conn()
+    if resource_group is None:
+        for group in list_resource_groups():
+            try:
+                instance_start = compconn.virtual_machines.start(group, name)
+                instance_start.wait()
+                resource_group = group

Same question about setting the resource group here as above.

—
You are receiving this because you were mentioned.
Reply to this email directly, view it on GitHub
<https://github.com/saltstack/salt/pull/45119#pullrequestreview-85363528>,
or mute the thread

**Comment by sumeetisp** [comment]
I have made the suggested changes

**Review by rallytime — APPROVED** [review]

**Comment by rallytime** [comment]
@sumeetisp This looks good, but there's a lint failure: https://jenkins.saltstack.com/job/PR/job/salt-pr-lint-n/17815/violations/file/salt/cloud/clouds/azurearm.py/

Once that is fixed up we can get this in. :)

**Comment by rallytime** [comment]
Go Go Jenkins!

**Pull request merged**